pkg/controller/dnsendpoint/nameserver/azure.go

```diff
@@ -181,3 +196,7 @@ func (q *azureQuery) queryNameServers(azureClient azureclient.Client, rootDomain
 
 	return nameServers, nil
 }
+
+func (q *azureQuery) getRelativeDomain(rootDomain string, domain string) string {
+	return strings.TrimRight(strings.TrimSuffix(domain, rootDomain), ".")
```

Comment:
To stay consistent with the other actuators, use `controllerutils.Undotted` instead of `strings.TrimRight`.
```suggestion
	return controllerutils.Undotted(strings.TrimSuffix(domain, rootDomain))
```